tools/GenerateXdmf.py

```diff
@@ -47,52 +47,100 @@ def generate_xdmf(file_prefix, output_filename, start_time, stop_time, stride):
         # loop over each h5 file
         for h5file in h5files:
             h5temporal = h5file[0].get('element_data.vol').get(id_and_value[0])
+            dimensionality = 3
+            # If there are no z-coordinates then assume the data is 2d. If
+            # in the future this assumption is invalid on datasets we will
+            # need an extra command line argument.
+            if not 'InertialCoordinates_z' in list(h5temporal.keys()):
```

Comment:
I wonder if for slices of 3D data this key will always be present. Can the current code handle slices through 3D data correctly?

Reply:
I have no (easy) way of testing that currently. I would expect something to break in that case.

Reply:
Does Spectre output 2D data that's not just (xcoords, ycoords)?  For example is there any function that would return a 2d sphere's worth of data?